.github/workflows/00_pull_request_entry_point.yml

```diff
@@ -0,0 +1,50 @@
+name: Pull Request Entry Point
+
+on:
+  pull_request:
+
+jobs:
+  check-permissions:
+    if: ${{ !contains(github.event.pull_request.labels.*.name, 'run-no-ci') }}
+    uses: ./.github/workflows/90_helpers_check-permissions.yml
+    with:
+      github-event-name: ${{ github.event_name }}
+
+  projects-changed:
+    needs: [ check-permissions ]
+    name: Check source files for changes
+    runs-on: ubuntu-22.04
+    timeout-minutes: 2
+    outputs:
+      go-projects-changes: ${{ steps.go-projects-changes.outputs.changes }}
+
+    steps:
+      - uses: actions/checkout@v4
+      - name: Go Projects Changes
+        id: go-projects-changes
+        uses: dorny/paths-filter@1441771bbfdd59dcd748680ee64ebd8faab1a242  #v3
+        with:
+          token: ${{ github.token }}
+          filters: .github/file-filters/go-build.yaml
+          base: ${{ github.event_name != 'pull_request' && (github.event.merge_group.base_ref || github.ref_name) || '' }}
+          ref: ${{ github.event_name != 'pull_request' && (github.event.merge_group.head_ref || github.ref) || '' }}
```

Comment:
we only run this on `pull_request`, so this seems a bit weird, but I guess that's for other repos primarily?

Reply:
it's not clear for me what will be the best final option:
1. to have same workflow for both `pull_request` and `push`
2. or to have separate workflows for both of them

for now it's only for `pull_request` events to simplify development process and debugging
at the same time I try to keep it the way so it easier to apply to both `pull_request` and `push` in the future.

in case it will be only for `pull_request`, we could just delete these 2 lines